### ISSUE TITLE
[Windows/Android][Movie/MPEG?][Glitch]Some games appear 3 vertical lines on the movie scean.

### ISSUE THREAD

**Issue by nassau-tk** [hrydgard/ppsspp#13697]
### What happens?
Some games appear 3 vertical lines on the movie scean.

Example)
- Muso Orochi (Not Include Musou Orochi Sairin & 2 Special)
- EXIT
- EXIT2 (Kangaeru EXIT)

PLease,See attached screenshots.

### What *should* happen?
These movies should draw  correctly.

### What hardware, operating system, and PPSSPP version?  On desktop, GPU matters for graphical issues.

PPSSPP Ver.：1.10.3-1234
OS：Windows10（64-bit）Japanese
CPU：Ryzen7 3800X（8-core）
GPU：RADEON RX5700XT
Resolution：2560×1440
RAM：32GB

PPSSPP Ver.：1.10.3-1234
OS：Android 10.0
DeviceName：Samusung Galaxy S20 / SC-51A
CPU：Snapdragon865（Octa-core）
GPU：Adreno 650
Resolution：3200 x 1440 (Max)
RAM：12GB
ROM：128GB
ExternalStrage：microSDXC 512GB

**[Muso Orochi]**
![muso_orochi_opening_movie](https://user-images.githubusercontent.com/48179091/99922248-ebc4d880-2d72-11eb-935d-fa2173c0d624.png)

**[EXIT]**
![EXIT_opening_movie](https://user-images.githubusercontent.com/48179091/99922257-fed7a880-2d72-11eb-860a-aeac26fa0926.png)

**[EXIT2]**
![EXIT2_opening_movie](https://user-images.githubusercontent.com/48179091/99922261-0a2ad400-2d73-11eb-950d-5b33fec859c7.png)

**Comment by Panderner** [comment]
Still happens in 1x resolution @nassau-tk?

**Comment by nassau-tk** [comment]
@Panderner 

Oops!!
I forgot to write an usual information.:smile:

1x Resolution is not appear these 3 lines.
It will appear 2x Resolution and more.

**Comment by hrydgard** [comment]
Yeah, very typical of an issue when a game is not written to expect higher resolution. It draws the video in four separate strips but not quite matching coordinates. We might be able to hack around it, but it'll be ugly.

**Comment by unknownbrackets** [comment]
At best we might be able to detect this is from a video and blit it directly...

-[Unknown]

**Comment by Panderner** [comment]
This also affects Fate/unlimited codes.
![Screenshot_2021-01-12-15-38-36-102_org ppsspp ppsspp](https://user-images.githubusercontent.com/37503397/104284052-621cc600-54ec-11eb-8404-399c2855ef11.jpg)

**Comment by Panderner** [comment]
and also Kamen Rider Super Climax Heroes too:
![Screenshot (14)](https://user-images.githubusercontent.com/37503397/115958527-56548d00-a53a-11eb-9a97-77fd6cbff5c2.png)

**Comment by unknownbrackets** [comment]
I moved the flush, DispatchSubmitImm does it now.  Shouldn't be lost.

-[Unknown]

**Comment by Panderner** [comment]
> Does the "Lazy texture caching" affect it?

Lazy texture caching had no effect for between v1.12.3-770-gc1b8fb737 and v1.12.3-776-g1c18c172a



> What game is that?

Fate: Unlimited Codes and Kamen Rider games (Climax Heroes OOO, Climax Heroes Fourze, and Super Climax Heroes)

**Comment by unknownbrackets** [comment]
Does it reproduce in a GE frame dump - yes would mean that a frame dump works in 770 but not in 776.  If so, that would be very useful.

DispatchSubmitImm() and Execute_End() are the only changes, so it ought to be related to one of those, but I only just moved the flushes around.  It already had FLUSHBEFORE...

-[Unknown]

**Comment by Panderner** [comment]
Here's a GE Dump:
[recording.zip](https://github.com/hrydgard/ppsspp/files/7972339/recording.zip)

and Screenshot for GE Dump:
![Screenshot (20)](https://user-images.githubusercontent.com/37503397/151831724-a038a705-8c56-43c4-9ec3-0ecf6a1a8cc3.png)

**Comment by hrydgard** [comment]
@unknownbrackets I didn't realize this before, but FLUSHBEFORE seems to be a no-op in FastRunLoop! Doh. Can't believe we never noticed. 

FLUSHBEFOREONCHANGE is implemented and works though, otherwise a lot of stuff would be broken.

**Comment by hrydgard** [comment]
I think it's actually right that we don't implement FLUSHBEFORE in FastRunLoop, don't want any additional check in there that we can live without, and since (except SIGNAL/FINISH) it's always used with EXECUTE, it can just be merged into the execution function.

Though SIGNAL/FINISH not flushing does seem suspicious, hehe

**Comment by hrydgard** [comment]
Should be fixed now, or at least only happening with skip buffer effects. Closing.

**Comment by Panderner** [comment]
Vertical lines still remains for increased internal resolutions.

<img width="1920" height="1088" alt="Image" src="https://github.com/user-attachments/assets/5cbe3fa2-1484-43e7-a893-8735fd13ea55" />

<img width="1920" height="1088" alt="Image" src="https://github.com/user-attachments/assets/cfd56414-5981-41e6-84c9-26bbe5c3e656" />
